### ISSUE TITLE
Running PandomiumTest never stop - Windows 10 64 bits

### ISSUE THREAD

**Issue by userquin** [Panda-Programming-Language/Pandomium#3]
Running the test, the application never exits. 

I think jcef app will hold a non daemon thread, preventing the jvm to halt.

**Comment by dzikoysk** [comment]
I've just changed a little disposing system, so try to run 1.0.1 version
```xml
<dependency>
    <groupId>org.panda_lang</groupId>
    <artifactId>pandomium</artifactId>
    <version>1.0.1</version>
</dependency>
```
If it works, you should see

![Console](https://panda-lang.org/screenshot/3g69bsMY.png)

**Comment by userquin** [comment]
know ot works with swing, but javafx still fails...

**Comment by dzikoysk** [comment]
Oh, currently you can't embed it in JavaFx because it's heavyweight component ([SwingNode](https://docs.oracle.com/javase/8/javafx/api/javafx/embed/swing/SwingNode.html)). I have to modify original jcef source.

I'm using JFXPanel to embed JavaFx in Swing in my application for now.

**Comment by userquin** [comment]
Running the sample first time from intellij, the app never stops (while stopping it using the red button, its icon is replaced with a skull icon and I must kill the java process from the task manager), second run works as expected (in the same intellij, killing or not killing the first run process). I will have to test with launch4j exe wrapper and see what happens on first execution.

**Comment by dzikoysk** [comment]
I have no idea why the jcef thread won't stop for you. Will the app close when you click the close button?

**Comment by userquin** [comment]
Upps, I'm running 1.0.1 version.

For your info, I'm testing pandomium with an app using http2 over TLS 1.2 with mutal authentication (X509 certificates and private keys: tls auth over https/1.1, http2 does not support it). The app is a PWA app built with Polymer, it is running as faster as running it on chrome, chromium or last safari. First time I run the app (from intellij), the app take around 5 seconds to show the splash of my app but next runs shows me the splash app very quickly, I dont know if is intellij or jcef. I think the problem is here, something is holding a non daemon thread, I keep waiting the process finish more than 1 hour, but never finish.

The JMV options are for integration on my project, consuming the vulcanized PWA app, I bundle a ZIP with all resources (ZIP NIO file system), and then intercept all calls using jcef app handler in my own scheme. I need to embed my app on windows tablets with no connectivity. I have the same project running also on android 5, 6 and 7, also on server side, with the same presentation layer built with Polymer (around 5000 html pages, so it is a huge project).

If you need some help about testing environments, such as linux or mac os x even also windows 10 I can help you, do not hesitate to contact me at userquin@gmail.com.

**Comment by userquin** [comment]
Whe I close the app using X window button, the frame is disposed, but the java process remain alive. If a stop using the red button intellij provides, then the skull icon appears, but cannot end the process, then I must kill it using task manager.

**Comment by userquin** [comment]
I have made a change to the test to show the frame in the EDT:

```java
        PandomiumSettings settings = PandomiumSettings.getDefaultSettings();

        Pandomium pandomium = new Pandomium(settings);
        pandomium.initialize();

        PandomiumClient client = pandomium.createClient();
        PandomiumBrowser browser = client.loadURL(args[0]);
        SwingUtilities.invokeLater(() -> {
            JFrame frame = new JFrame();
            frame.getContentPane().add(browser.toAWTComponent(), BorderLayout.CENTER);
            frame.pack();

            frame.setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
            frame.addWindowListener(new WindowAdapter() {
                @Override
                public void windowClosing(WindowEvent e) {
                    pandomium.dispose();
                    frame.dispose();
                }
            });

            frame.setTitle("Pandomium");
            frame.setSize(1380, 760);
            frame.setVisible(true);
        });
```

**Comment by dzikoysk** [comment]
I'm trying to exclude the jcef thread-logic to the PandomiumThread. We will see if manual shutdown would work

**Comment by userquin** [comment]
It is very strange, the error is spurious. Closing all intellij instances I cannot reproduce the error, It seems to be on pc startup or restart, so I think maybe it is a problem with intellij. I have reproduced twice, yesterday evening and today.

Do not change the code, I have to finish some tasks and then I'll build an exe for windows to see if I can reproduce the problem, I'll keep you informed...

**Comment by userquin** [comment]
I have restarted my windows 10 pc and the problem is here again, using panda-soft url in the first execution, this is what debug.log contains:
![error](https://cloud.githubusercontent.com/assets/6311119/26756111/e3adb636-489b-11e7-9786-edb302f2e89b.png)

and debug.log contains:

[0603/202628.191:ERROR:child_thread_impl.cc(762)] Request for unknown Channel-associated interface: ui::mojom::GpuMain
[0603/202628.316:ERROR:gpu_child_thread.cc(330)] Exiting GPU process due to errors during initialization

My windows PC is a i7 with Nvidia 840M

**Comment by userquin** [comment]
If I run again the app, this second one will not reproduce the error, closing the jvm process also with the first one yet running:

![errro2](https://cloud.githubusercontent.com/assets/6311119/26756238/f07c4e0c-489d-11e7-8828-b83f286d6862.png)

**Comment by dzikoysk** [comment]
I'll add jcef sources to the pandomium module tomorrow and I'll modify it finally. It looks like jcef bug because it posts shutdown hook to stop the cefapp. Unfortunately, the environment does not always call shutdown hooks :/

Anyway, I'm using windows 10 x64

**Comment by userquin** [comment]
It will posible to make this feature configurable? Let say, a method in some part other than jcef or pandomium thread? 

Suppose we have a swing app with 100 screens, but only 1 shows jcef? how can we dispose jcef resources to free all its resources? My app do not have 100 screens but it has a few outside jcef, for example the installer, the configurer, sync options, backup, report a bug... jcef is just one screen. Once jcef is started we need a way to shutdown or start it again, based on user interaction...

**Comment by userquin** [comment]
Let's go from bad to worse, this morning the windows computer only with the intellij started and with the application of pandomium gives me the famous blue screen of windows, there must be a serious problem with the native part.

Now Intellij gets corrupted, I need to reinstall it.

![blue_screen](https://cloud.githubusercontent.com/assets/6311119/26761629/826b39b4-4933-11e7-9aac-72fb4aa4ffd0.jpg)

![intellij](https://cloud.githubusercontent.com/assets/6311119/26761630/88e84c50-4933-11e7-88a5-419b449f2f4e.jpg)

**Comment by dzikoysk** [comment]
I've just published 1.0.3 version with modified jcef sources. 
If the problem exists in the native code then I can't fix it :/
